### DISCUSSION TITLE
Issue with Converting .inp File to .msh File for Neper

### DISCUSSION THREAD

**Discussion by nippyawn** [neperfepx/neper#956] [Q&A] [answered]
Hello,

I am trying to convert an .inp file into an .msh file. I rewrote the .inp file following the .msh format rules. The converted file, named test_30.msh, is attached.

When I deleted everything below the $Periodicity section in a .msh file that was originally created by Neper using the normal procedure, the .msh file worked correctly.

However, when I try to load this file into Neper, I encounter an error that I cannot resolve. Below is the command I used and the error message:

The $Elements section of the converted file is structured as follows:
<elt_id> <elt_type> <number_of_tags> <tag1> ... <elt_id_node1> ...

<elt_type>: I use 4 because my mesh is constructed with C3D4 elements.
<tag1,2>: These are the elset IDs.

````
user@Ubuntu:~/root/1219_test$ neper -M -loadmesh test_30.msh -o 1

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.9.1-1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 20 threads.
Info   : <https://neper.info/>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/user/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -M loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -loadmesh test_30.msh -o 1
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   :   - Reading arguments...
Info   : Loading input data...
Info   : Loading result mesh...
Info   :     [i] Parsing file `test_30.msh'...
Info   :     [i] Parsed file `test_30.msh'.
Info   :   - Reconstructing mesh...
Info   :     > Reconstructing 2D mesh... 100%      
Info   :     > Reconstructing 1D mesh... 100%      
Info   :     > Reconstructing 0D mesh... 100%      
Error  : You have discovered a bug in Neper!  Please file an issue at
         https://github.com/rquey/neper/issues.  Thank you.

中止
````
I would greatly appreciate any advice or help in resolving this issue. Thank you!
[test_30.txt](https://github.com/user-attachments/files/18194106/test_30.txt)

**Comment by rquey** [comment, marked as answer]
Add, right before the `$Nodes` section:
```
$Topology
0
$EndTopology
```
and use the latest Neper version (4.10.2-6).
